### PR TITLE
Solve a stack_use_after_scope issue in sockaddr_resolver_test

### LINKED ISSUE #14075: ASAN error in sockaddr_resolver_test
```
oot@4daff6cf41c9:/grpc# cat /root/.cache/bazel/_bazel_root/beca832d8582c7cdc0e632cac0ff1dfc/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/testlogs/test/core/client_channel/resolvers/sockaddr_resolver_test@poller=poll-cv/test.log
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
-----------------------------------------------------------------------------
+ export GRPC_POLL_STRATEGY=poll-cv
+ shift
+ test/core/client_channel/resolvers/sockaddr_resolver_test
D0118 21:06:13.092926281   47965 test_config.cc:388]         test slowdown factor: sanitizer=3, fixture=1, poller=5, total=15
E0118 21:06:13.094130273   47965 parse_address.cc:79]        no port given for ipv4 scheme
=================================================================
==47965==ERROR: AddressSanitizer: stack-use-after-scope on address 0x7ffebf052498 at pc 0x0000004eebe1 bp 0x7ffebf0520b0 sp 0x7ffebf0520a8
READ of size 8 at 0x7ffebf052498 thread T0
    #0 0x4eebe0 in on_resolution_cb(void*, grpc_error*) /proc/self/cwd/test/core/client_channel/resolvers/sockaddr_resolver_test.cc:40:34
    #1 0x4f0415 in closure_impl::closure_wrapper(void*, grpc_error*) /proc/self/cwd/./src/core/lib/iomgr/closure.h:150:3
    #2 0x7f7e2a6fa03e in exec_ctx_run(grpc_closure*, grpc_error*) /proc/self/cwd/src/core/lib/iomgr/exec_ctx.cc:38:3
    #3 0x7f7e2a6f9c17 in grpc_core::ExecCtx::Flush() /proc/self/cwd/src/core/lib/iomgr/exec_ctx.cc:125:9
    #4 0x4eff46 in grpc_core::ExecCtx::~ExecCtx() /proc/self/cwd/./src/core/lib/iomgr/exec_ctx.h:85:5
    #5 0x4efa62 in test_succeeds(grpc_resolver_factory*, char const*) /proc/self/cwd/test/core/client_channel/resolvers/sockaddr_resolver_test.cc:68:1
    #6 0x4eed6a in main /proc/self/cwd/test/core/client_channel/resolvers/sockaddr_resolver_test.cc:97:3
    #7 0x7f7e293fbb44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b44)
    #8 0x4211eb in _start (/root/.cache/bazel/_bazel_root/beca832d8582c7cdc0e632cac0ff1dfc/execroot/com_github_grpc_grpc/bazel-out/k8-fastbuild/bin/test/core/client_channel/resolvers/sockaddr_resolver_test+0x4211eb)

Address 0x7ffebf052498 is located in stack of thread T0 at offset 216 in frame
    #0 0x4ef50f in test_succeeds(grpc_resolver_factory*, char const*) /proc/self/cwd/test/core/client_channel/resolvers/sockaddr_resolver_test.cc:43

  This frame has 3 object(s):
    [32, 104) 'exec_ctx' (line 44)
    [144, 176) 'args' (line 46)
    [208, 224) 'on_res_arg' (line 57) <== Memory access at offset 216 is inside this variable
HINT: this may be a false positive if your program uses some custom stack unwind mechanism or swapcontext
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-use-after-scope /proc/self/cwd/test/core/client_channel/resolvers/sockaddr_resolver_test.cc:40:34 in on_resolution_cb(void*, grpc_error*)
Shadow bytes around the buggy address:
  0x100057e02440: 00 00 00 00 f1 f1 f1 f1 00 00 f3 f3 00 00 00 00
  0x100057e02450: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100057e02460: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100057e02470: 00 00 00 00 00 00 00 00 f1 f1 f1 f1 00 00 00 00
  0x100057e02480: 00 00 00 00 00 f2 f2 f2 f2 f2 f8 f8 f8 f8 f2 f2
=>0x100057e02490: f2 f2 f8[f8]f3 f3 f3 f3 00 00 00 00 00 00 00 00
  0x100057e024a0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100057e024b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100057e024c0: 00 00 00 00 f1 f1 f1 f1 f8 f8 f8 f8 f8 f8 f8 f8
  0x100057e024d0: f8 f3 f3 f3 f3 f3 f3 f3 00 00 00 00 00 00 00 00
  0x100057e024e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==47965==ABORTING
```

### PR DESCRIPTION
**Pull request by yashykt**
Fixes #14075

### CODE REVIEW

File: test/core/client_channel/resolvers/sockaddr_resolver_test.cc [hunk 1]
```diff
@@ -63,8 +63,10 @@ static void test_succeeds(grpc_resolver_factory* factory, const char* string) {
   grpc_resolver_next_locked(resolver, &on_res_arg.resolver_result,
                             on_resolution);
   GRPC_RESOLVER_UNREF(resolver, "test_succeeds");
-
   grpc_uri_destroy(uri);
+  /* Flush ExecCtx to avoid stack-use-after-scope on on_res_arg which is
+   * accessed in the closure on_resolition_cb */
```
Comment (dgquintas) [inline test/core/client_channel/resolvers/sockaddr_resolver_test.cc]: s/resolition/resolution

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: test/core/client_channel/resolvers/sockaddr_resolver_test.cc (hunk 1 above)
Comment (yashykt) [reply test/core/client_channel/resolvers/sockaddr_resolver_test.cc]: Done!

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by yashykt** [comment]
Known Issues : #13149

**Pull request merged**